subscription/src/main/java/org/killbill/billing/subscription/engine/dao/DefaultSubscriptionDao.java

```diff
@@ -469,22 +469,28 @@ public DefaultSubscriptionBase apply(@Nullable final SubscriptionModelDao input)
             @Override
             public List<DefaultSubscriptionBase> inTransaction(final EntitySqlDaoWrapperFactory entitySqlDaoWrapperFactory) throws Exception {
 
-                final List<SubscriptionBundleModelDao> bundleModels = entitySqlDaoWrapperFactory.become(BundleSqlDao.class).getByAccountRecordId(context);
 
                 final SubscriptionSqlDao subscriptionSqlDao = entitySqlDaoWrapperFactory.become(SubscriptionSqlDao.class);
                 final List<SubscriptionModelDao> subscriptionModels = cutoffDt == null ?
                                                                       subscriptionSqlDao.getByAccountRecordId(context) :
                                                                       subscriptionSqlDao.getActiveByAccountRecordId(cutoffDt.toDate(), context);
+
+                // We avoid pulling the bundles when a cutoffDt is specified, as those are not really used
```

Comment:
We could also issue a separate query, specifically to pull these external keys?

Reply:
That's a good point, let's discuss...